src/storage/mod.rs

```diff
@@ -1847,47 +1847,34 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
         }
     }
 
-    fn get_causal_ts(ts_provider: &Option<Arc<CausalTsProviderImpl>>) -> Result<Option<TimeStamp>> {
-        if let Some(p) = ts_provider {
-            match p.get_ts() {
-                Ok(ts) => Ok(Some(ts)),
-                Err(e) => Err(box_err!("Fail to get ts: {}", e)),
-            }
-        } else {
-            Ok(None)
-        }
-    }
+    async fn check_causal_ts_synced(ctx: &mut Context, api_version: ApiVersion) -> Result<()> {
+        if api_version == ApiVersion::V2 {
```

Comment:
`F::TAG` can be utilized to avoid pass the `api_version` argument.

Reply:
I used F::TAG instead of argument, thanks.